pkg/proxy/iptables/proxier.go

```diff
@@ -135,6 +135,37 @@ func (lkct LinuxKernelCompatTester) IsCompatible() error {
 const sysctlRouteLocalnet = "net/ipv4/conf/all/route_localnet"
 const sysctlBridgeCallIPTables = "net/bridge/bridge-nf-call-iptables"
 
+// Since converting probabilities (floats) to strings is expensive
+// and we are using only probabilities in the format of 1/n, we are
+// precomputing some number of those and cache for future reuse.
+var precomputedProbabilities []string
+
+func init() {
+	const initiallyPrecomputed = 1000
+	precomputedProbabilities = make([]string, 0, initiallyPrecomputed+1)
+	precomputeProbabilities(initiallyPrecomputed)
+}
+
+func precomputeProbabilities(numberOfPrecomputed int) {
+	if len(precomputedProbabilities) == 0 {
```

Comment:
Introducing mutex will most probably eliminate the gain from precomputing. Because we would need to protect the whole "precomputedProabilities" field.
Currently, it's not a problem, because all calls to it are from syncProxyRules so under  Proxier.lock.

@thockin - are you fine with just adding a comment that this is protected by Proxier.lock? Or maybe should make this a field of proxier?